parser/src/powdr.lalrpop

```diff
@@ -622,6 +633,22 @@ IfExpression: Box<Expression> = {
         <end:@R> => ctx.to_expr_with_source_ref(IfExpression{condition, body, else_body}, start, end),
 }
 
+StructExpression: Box<Expression> = {
+    <start:@L> <name:Identifier> "with" "{" <fields:StructExpressionList> "}" <end:@R> => ctx.to_expr_with_source_ref(StructExpression{name, fields}, start, end),
+}
+
+StructExpressionList: BTreeMap<String, Box<Expression>> = {
```

Comment:
I wonder if we should use `vec` here instead. If we re-format, BTreeMap sorts the fields.
I think as long as we use BTreeMap for the Values in the evaluator, it should not be bad WRT performance, what do you think?

Reply:
I don't think having a vector here will create a problem. In fact, I went for a BTreeMap mainly because it seemed the most "ergonomic" option to store expressions associated to strings. I'll change it :)